app/helpers/repository_helper.rb

```diff
@@ -105,6 +105,10 @@ def remove_files(files, user, repo, path: '/', txn: nil, keep_folder: true)
     end
 
     current_path = Pathname.new path
+    relative_path = Pathname.new(files[0])
+    abs_path = File.join(current_path, relative_path)
+
+    return [true, [:file_not_exist]] unless repo.get_latest_revision.path_exists?(abs_path)
```

Comment:
This is only checking if the first file in `files` doesn't exist. What happens if there are multiple files in `files` and the second, third, etc. file doesn't exist?

Reply:
Good point.Here in remove_folder (submission_controller)we can delete the files one by one not as a list. So if any other method which access remove_folder in repository_helper sends multiple files that time it wont work. I will make sure this works for all the methods which is accessing remove_folder method. Thanks for pointing this.

Reply:
*remove_file